ttyclock.c

```diff
@@ -679,11 +678,89 @@ main(int argc, char **argv)
           update_hour();
           draw_clock();
           key_event();
+          ttyclock.lt = timestamp();
      }
 
      endwin();
 
      return 0;
 }
 
+time_t timestamp(void)
+{
+     if (whole_seconds_delay())
+     {
+          return rounded_timestamp();
+     }
+     else
+     {
+          return time(NULL);
+     }
+}
+
+time_t rounded_timestamp(void)
+{
+     static const long HALF_SECOND_IN_NANOSECONDS = 500 * 1000 * 1000;
+     struct timespec currentTime;
+
+     if (clock_gettime(CLOCK_REALTIME, &currentTime) != 0)
+     {
+          return time(NULL);
+     }
+
+     if (currentTime.tv_nsec >= HALF_SECOND_IN_NANOSECONDS)
+     {
+          return currentTime.tv_sec + 1;
+     }
+     else
+     {
+          return currentTime.tv_sec;
+     }
+}
+
+struct timespec sleep_length(void)
+{
+     static const long SECOND_IN_NANOSECONDS = 1000 * 1000 * 1000;
+     static const long NINE_TENTHS = 900 * 1000 * 1000;
+
+     if (whole_seconds_delay())
+     {
+          struct timespec currentTime;
+
+          if (clock_gettime(CLOCK_REALTIME, &currentTime) == 0 &&
+               currentTime.tv_nsec > 0)
+          {
+               struct timespec result;
+
+               // avoiding sleeping for < 1 / 10 of a second
```

Comment:
nice try, but "why?" why do we avoid sleeping for a tenth? not worth it? how do we handle it in that case?

Reply:
Will amend the comment